be/src/olap/rowset/segment_v2/inverted_index_writer.cpp

```diff
@@ -394,9 +394,11 @@ class InvertedIndexColumnWriterImpl : public InvertedIndexColumnWriter {
                             _parser_type != InvertedIndexParserType::PARSER_NONE) {
                             // in this case stream need to delete after add_document, because the
                             // stream can not reuse for different field
-                            _char_string_reader->init(v->get_data(), v->get_size(), false);
+                            std::unique_ptr<lucene::util::Reader> char_string_reader = nullptr;
```

Comment:
because if we reuse the iobuffer in TokenStream will be replaced with last element in current array which make fields in current doc has same value (the last element).